manifests/repo.pp

```diff
@@ -32,15 +32,14 @@
 
   case $::osfamily {
     'Debian': {
-      if !defined(Class['apt']) {
-        class { 'apt': }
-      }
+      include ::apt
+      Class['apt::update'] -> Package<| tag == 'elasticsearch-package' |>
```

Comment:
Even though something like this would ideally be handled by the `apt` module, requiring `apt::update` before packages is a [known limitation](https://github.com/puppetlabs/puppetlabs-apt#limitations). Using tags on the package resource should avoid help avoid dependency cycles since the collector is limited to packages we install.